AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs

```diff
@@ -21,14 +24,40 @@ public class PerformanceCollectorModule : ITelemetryModule
                                                                 @"\Processor(_Total)\% Processor Time"
                                                             };
 
-        public IList<PerformanceCounterCollectionRequest> Counters { get; private set; }
+        public List<PerformanceCounterCollectionRequest> Counters { get; private set; }
 
+        /// <summary>
+        /// Initializes the default performance counters.
+        /// </summary>
         public void Initialize(TelemetryConfiguration configuration)
         {
             CounterFactory factory = new CounterFactory();
 
             foreach (string counter in defaultCounters)
-                factory.GetCounter(counter);
+            {
+                try
+                {
+                    ICounterValue c = factory.GetCounter(counter);
+                    configuration.RegisterCounter(c);
+                }
+                catch (ArgumentException e)
```

Comment:
this is strange to catch and rethrow exception without any other code or comment